pysph/sph/wc/linalg.py

```diff
@@ -62,6 +112,115 @@ def mat_vec_mult(a=[1.0, 0.0], b=[1.0, 0.0], n=3, result=[0.0, 0.0]):
         result[i] = s
 
 
+def linear_solver_2d(A=[0.0, 0.0], b=[0.0, 0.0], result=[0.0, 0.0]):
+    """Linear solver using Cramers's rule
+
+    Given flattened matrix `A` and vector `b`. Cramer's rule is used to
+    determine the result. In case the deteminant is zero, values considering
+    the diagonal terms is returned. Doing this is equivalent to Shepard
+    interpolation in the context of SPH.
+
+    Parameters
+    -----------
+    A: list: given matrix.
+    b: list: given vector
+    result: list: x = inv(A)b
+    """
+
+    detA = A[0]*A[3] - A[1]*A[2]
+    if abs(detA) > 1e-14:
+        detA1 = b[0]*A[3] - A[1]*b[1]
+        detA2 = A[0]*b[1] - b[0]*A[2]
+        result[0] = detA1/detA
+        result[1] = detA2/detA
+    else:
+        result[0] = b[0]/A[0]
+        result[1] = b[1]/A[3]
```

Comment:
Same goes here! `A[3]` or others may not be initialized properly. Also, the else part is not clear to me, why divide by the digonal elemets, what if the determinant is zero as well as the diagonal elements.

Reply:
The else part is specific to SPH, the determinant could be zero due to lack of support in those cases the result will be shepard interpolation.